lib/eclipse/tests/ScheduleTests.cpp

```diff
@@ -2178,3 +2178,40 @@ BOOST_AUTO_TEST_CASE(handleWEFAC) {
     BOOST_CHECK_EQUAL(well_i->getEfficiencyFactor(3), 0.9);
 
 }
+
+BOOST_AUTO_TEST_CASE(historic_BHP_and_THP) {
+    Opm::Parser parser;
+    std::string input =
+        "START             -- 0 \n"
+        "19 JUN 2007 / \n"
+        "SCHEDULE\n"
+        "DATES             -- 1\n"
+        " 10  OKT 2008 / \n"
+        "/\n"
+        "WELSPECS\n"
+        " 'P' 'OP' 9 9 1* 'OIL' 1* / \n"
+        " 'I' 'OP' 9 9 1* 'WATER' 1* / \n"
+        "/\n"
+        "WCONHIST\n"
```

Comment:
Add a `WCONPROD` keyword with well `P2`- set the `BHP` and `THP` items of that well, and verify that the `BHPH` and `THPH` variables have *not* been set.